catalog/async/catalog-async-data/src/main/java/org/codice/ddf/catalog/async/data/impl/LazyProcessResourceImpl.java

```diff
@@ -0,0 +1,193 @@
+     * @throws IllegalArgumentException if the input provided is not valid
+     *
+     **/
+    public LazyProcessResourceImpl(String metacardId, Supplier<Resource> resourceSupplier) {
+        if (StringUtils.isBlank(metacardId)) {
+            throw new IllegalArgumentException(
+                    "ProcessResourceImpl argument \"metacardId\" may not be null or empty.");
+        }
+        if (resourceSupplier == null) {
+            throw new IllegalArgumentException(
+                    "LazyProcessResourceImpl must have a non null resource supplier");
+        }
+
+        this.isModified = false;
+        this.metacardId = metacardId;
+        this.resourceSupplier = resourceSupplier;
+        this.isResourceLoaded = false;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * This will load the resource.
+     */
+    @Override
+    public InputStream getInputStream() throws IOException {
+        loadResource();
+        return inputStream;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * This will load the resource.
+     */
+    @Override
+    public String getMimeType() {
+        loadResource();
+        return mimeType;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * This will load the resource.
+     */
+    @Override
+    public String getName() {
+        loadResource();
+        return name;
+    }
+
+    @Override
+    public String getQualifier() {
+        return qualifier;
+    }
+
+    @Override
+    public long getSize() {
+        return size;
+    }
+
+    @Override
+    public URI getUri() {
+        return uri;
+    }
+
+    public void setSize(long size) {
+        this.size = size;
+    }
+
+    public void setUri(URI uri) {
+        this.uri = uri;
+
+        if (uri != null && StringUtils.isNotBlank(uri.getFragment())) {
+            this.qualifier = uri.getFragment();
+        }
+    }
+
+    @Override
+    public boolean isModified() {
+        return isModified;
+    }
+
+    @Override
+    public void close() {
+        IOUtils.closeQuietly(inputStream);
+    }
+
+    public void markAsModified() {
+        isModified = true;
+    }
+
+    private void loadResource() {
+        if (!isResourceLoaded) {
+            populateProcessResource(resourceSupplier.get());
+        }
+    }
+
+    private void populateProcessResource(Resource resource) {
+        if (resource == null) {
+            LOGGER.debug(
+                    "Error loading resource for metacard id: {}. null resource was returned by supplier.",
+                    metacardId);
+            return;
+        }
+
+        this.inputStream = resource.getInputStream();
+
+        if (this.size == UNKNOWN_SIZE) {
+            this.size = resource.getSize();
+        }
+
+        String resourceName = resource.getName();
+        if (StringUtils.isNotBlank(resourceName)) {
+            this.name = resourceName;
+        } else {
+            this.name = DEFAULT_NAME;
```

Comment:
Why not instantiate `name` and `mimeType` to `DEFAULT_NAME` and `DEFAULT_MIME_TYPE` like we do for `size`:
```java
private long size = UNKNOWN_SIZE;
```

What happens for  `getMimeType()` and `getName()` when we are unable to load the resource? The javadoc does not allow for null mimetype or name.

Reply:
Changed to default mimeType and name.